frame/election-provider-multi-phase/src/signed.rs

```diff
@@ -395,10 +395,20 @@ impl<T: Config> Pallet<T> {
 		// Any unprocessed solution is pointless to even consider. Feasible or malicious,
 		// they didn't end up being used. Unreserve the bonds.
 		let discarded = all_submissions.len();
-		for SignedSubmission { who, deposit, .. } in all_submissions.drain() {
+		let mut count = 0;
+		for SignedSubmission { who, deposit, call_fee, .. } in all_submissions.drain() {
+			if T::SignedMaxRefunds::get().map_or(true, |max| count < max) {
```

Comment:
You are doing one map access here per loop, you can probably save that in a temp to lower the weight.